### PR TITLE
[Enhancement]improve err msg of noexist key column

### LINKED ISSUE #7719: Error message is not good when creating a table with a nonexistent column
## Enhancement

```SQL
CREATE TABLE t_tf (
    a int not null,
    b float,
    c1 varchar(100)
) DUPLICATE KEY (id)
DISTRIBUTED BY HASH(id) BUCKETS 1;
```

To create a table as above, but `id` does not exist.
However, the error message is:
`ERROR 1064 (HY000) at line 1: Key columns should be a ordered prefix of the schema.`

1. the message is not correct.
2. it should display the column name `id` does not exist.

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] enhancement

## Which issues of this PR fixes ：
Fixes #7719

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/KeysDesc.java|0|2|00.00%|

**Comment by imay** [comment]
@zombee0 you can comment related issues in commit message. like
Fixes #7719

File: fe/fe-core/src/main/java/com/starrocks/analysis/KeysDesc.java [hunk 1]
```diff
@@ -78,6 +78,9 @@ public void analyze(List<ColumnDef> cols) throws SemanticException {
         for (int i = 0; i < keysColumnNames.size(); ++i) {
             String name = cols.get(i).getName();
             if (!keysColumnNames.get(i).equalsIgnoreCase(name)) {
+                if (!cols.contains(keysColumnNames.get(i))) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/KeysDesc.java:81]: cols is a List<ColumnDef>, and kyesColumNames is a String.
how does `cols.contains` work.

And column should be case insensitive, does it work?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/analysis/KeysDesc.java:81]: the keyColumnNames is a list of string，get(i) will return a string，and I had test this code by building and testing the FE. I forgot the case insentivity，I will fix it later，thank you!

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/analysis/KeysDesc.java]: > cols is a List, and kyesColumNames is a String. how does `cols.contains` work.
> 
> And column should be case insensitive, does it work?

sorry, I read the code again, You are right, cols.contains not work, i made a mistake, and i had fixed it, thank you!

**Comment by zombee0** [comment]
I have several tests:

mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (id) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key column(id) doesn't exist.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (B) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key columns should be a ordered prefix of the schema.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (C1) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key columns should be a ordered prefix of the schema.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (C) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key column(C) doesn't exist.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (c1) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key columns should be a ordered prefix of the schema.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (c) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Key column(c) doesn't exist.
mysql> CREATE TABLE t_tf (a int not null, b float, c1 varchar(100) ) DUPLICATE KEY (A) DISTRIBUTED BY HASH(id) BUCKETS 1 PROPERTIES ("replication_num"="1");
ERROR 1064 (HY000): Distribution column(id) doesn't exist.
mysql>

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/KeysDesc.java|0|3|00.00%|

**Review by imay — APPROVED** [review]

**Pull request merged**